flambe/nn/module.py

```diff
@@ -120,3 +122,44 @@ def clip_gradient_norm(self, threshold: float):
         # Only compute over parameters that are being trained
         parameters = filter(lambda p: p.requires_grad and p.grad is not None, self.parameters())
         nn.utils.clip_grad_norm_(parameters, threshold)
+
+
+class Encoder(Module):
+    """A simple interface providing the input and output dimensions.
+
+    This class may be expanded in the future to perform type checking
+    on input, output dimensions as well as the number of dimensions.
+    The interface currently requires implementing two methods:
+
+    -  ``input_dim``:
+        The size of the last dimensions of an input
+    -  ``output_dim``:
+        The size of the last dimensions of an output
+
+    """
+
+    @abstractmethod
+    @property
+    def input_dim(self) -> int:
+        """Get the size of the last dimension of an input.
+
+        Returns
+        -------
+        int
+            The size of the last dimension of an input.
+
+        """
+        pass
```

Comment:
do we want to `pass` here or `raise NotImplementedError`? Same below

Reply:
always pass on abstractmethods